ordinal-encoder/metadata.json

```diff
@@ -0,0 +1,32 @@
+{
+    "name": "Ordinal encoder.",
+    "description": "Given a dataset, encodes categorical fields using ordinal ecoding, which uses a single column of integers to represent field classes (levels).",
+    "kind": "script",
+    "source_code": "script.whizzml",
+    "inputs": [
+        {
+            "name": "input-dataset-id",
+            "type": "dataset-id",
+            "description": "Dataset to be encoded."
+        },
+        {
+            "name": "target-fields",
+            "type": "list",
+            "default": [],
+            "description": "[Optional] A list of field names or identifiers to encode. Non-categorical fields are ignored. If not supplied, or an empty list, all categorical columns are encoded."
+        },
+        {
+            "name": "mappings",
+            "type": "map",
+            "default": {},
+            "description": "[Optional] A map for the encoding of each field. Top-level keys are field names or identifiers. Values are maps of classes to integers. For example, {'col1': {'A': 0, 'B': 1, 'C': 2}}. If no mapping is given for a field, classes are assumed to have no true order and integers are assigned at random. An incomplete mapping for a field causes missing values in the new (ordinal) field."
+        }
```

Comment:
classes do have a "natural" order, at least one that we use everywhere else: their order in the summary.categories list of the field descriptor.  we should use that order to assign consecutive integers, if no other is given.

Reply:
yes, "random" is incorrect. it's the order the classes are returned from the field-categories command. is that just hash order? it does not appear to be alphabetic, or the order in which they first appear, etc